x-pack/plugins/maps/public/routes/map_page/map_app/map_app.tsx

```diff
@@ -244,9 +244,17 @@ export class MapApp extends React.Component<Props, State> {
     } else {
       indexPatterns = await getIndexPatternsFromIds(nextIndexPatternIds);
     }
-    if (this._isMounted) {
-      this.setState({ indexPatterns });
+    
+    if (!this._isMounted) {
+      return;
+    }
+
+    // ignore results for outdated requests
+    if (!_.isEqual(nextIndexPatternIds, this._prevIndexPatternIds)) {
```

Comment:
This is an unrelated fix for an issue where timing of getIndexPatterns response were out of order and resulted in setting state to a stale value.